src/main.rs

```diff
@@ -49,9 +48,9 @@ lazy_static::lazy_static! {
     };
 }
 
-futures::task_local! {
+std::thread_local! {
```

Comment:
Why change the task local into a thread local?

Reply:
I mentioned this on matrix but no one really got back to me on it. Ill copy paste it here so its more open:

since `futures::task_local!` does not exist in `futures` 0.3 I'm using `tokio::task_local!`, but there is one notable difference in that you no longer set the default value:

```
futures::task_local!{
    static CONTEXT: RefCell<Option<ConnectionContext>> = RefCell::new(None);
}
```
becomes:

```
tokio::task_local!{
    static CONTEXT: RefCell<Option<ConnectionContext>>;
}
```
Running the tests from my branch here gives the error:

```
'cannot access a Task Local Storage value without setting it via `LocalKey::set`: AccessError'
```
but I dont see a `LocalKey::set` anywhere in the `tokio` docs. It looks like in [the source code](https://docs.rs/tokio/0.2.21/src/tokio/task/task_local.rs.html#34-46) `tokio` uses `std::thread_local!`, is it safe to use this as a replacement?